dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,325 @@
+            list(completed_jobs.values()))
+
+
+class HyperbandCV(DaskBaseSearchCV):
+    def __init__(self, model, params, max_iter=81, eta=3, asynchronous=None,
+                 random_state=42, scoring=None):
+        self.model = model
+        self.params = params
+        self.max_iter = max_iter
+        self.eta = eta
+        self.random_state = random_state
+        self.scoring = scoring
+        if asynchronous is None:
+            asynchronous = False
+        self.asynchronous = asynchronous
+
+        super(HyperbandCV, self).__init__(model)
+
+    def fit(self, X, y, **fit_params):
+        return default_client().sync(self._fit, X, y, **fit_params)
+
+    @gen.coroutine
+    def _fit(self, X, y, **fit_params):
+        if isinstance(X, np.ndarray):
+            X = da.from_array(X, chunks=X.shape)
+        if isinstance(y, np.ndarray):
+            y = da.from_array(y, chunks=y.shape)
+        r = yield _hyperband(self.model, self.params, X, y,
+                             max_iter=self.max_iter,
+                             asynchronous=self.asynchronous,
+                             fit_params=fit_params, eta=self.eta,
+                             random_state=self.random_state)
+        params, model_meta_futures, history, all_models, jobs = r
+
+        self.jobs_ = jobs
+        self.history_ = history
+        self._model_meta_futures = model_meta_futures
+        self.__all_models = all_models
+
+        cv_res, best_idx = _get_cv_results(jobs, params)
+        best_id = cv_res['model_id'][best_idx]
+
+        best_model_and_meta = yield model_meta_futures[best_id]
+        self.best_estimator_ = best_model_and_meta[0]
+
+        self.cv_results_ = cv_res
+        self.best_index_ = best_idx
+        self.n_splits_ = 1  # TODO: increase this! It's hard-coded right now
+        self.multimetric_ = False
+
+        return self
+
+    @property
+    def _models_and_meta(self):
+        return default_client().gather(self._model_meta_futures)
+
+    @property
+    def _all_models(self):
+        models_and_meta = default_client().gather(self.__all_models)
+        models = {k: [vi[0] for vi in v] for k, v in models_and_meta.items()}
+        return models
+
+    def info(self, jobs=None):
+        res = default_client().sync(self._info, jobs=jobs)
+        assert isinstance(res, dict)
+        return res
+
+    @gen.coroutine
+    def _info(self, jobs=None):
+        if jobs is None:
+            X, y = make_classification(n_samples=10, n_features=4, chunks=10,
+                                       random_state=self.random_state)
+            r = yield _hyperband(self.model, self.params, X, y,
+                                 max_iter=self.max_iter,
+                                 asynchronous=self.asynchronous,
+                                 dry_run=True, eta=self.eta,
+                                 random_state=self.random_state)
+            _, _, _, _, jobs = r
```

Comment:
I don't know this algorithm well, but this particular function smells odd to me.  I wouldn't mind seeing it and the dry_run stuff removed if possible